src/store/modules/geometry/actions/actions.js

```diff
@@ -94,17 +24,25 @@ export default {
       }, { root: true });
     },
 
-    /*
-    * Erase the selection defined by a set of points on all faces on the current story
-    * Dispatched by the eraser tool
-    */
-    eraseSelection (context, payload) {
-  		const { points } = payload;
-  		const eraseResult = eraseSelection(points, context);
+  /*
+  * Erase the selection defined by a set of points on all faces on the current story
+  * Dispatched by the eraser tool
+  */
+  eraseSelection(context, payload) {
+    const
+      { points } = payload,
+      geometry_id = context.rootGetters['application/currentStoryGeometry'].id;
+
+    withPreservedComponents(context, geometry_id, () => {
```

Comment:
So happy with this code I wrote a blog post about it https://brianschiller.com/blog/2017/09/20/python-style-context-managers-in-js